gfop/get_food_counts.py

```diff
@@ -0,0 +1,134 @@
+        sample_types_selected.isin(sample_counts_valid)]
+    # Get sample counts at the specified level.
+    return sample_types_selected.value_counts()
+
+
+def get_dataset_food_counts(gnps_network, metadata, filename_col,
+                            sample_types, all_groups, some_groups,
+                            level, save_name):
+    """
+    Generate a table of food counts for a study dataset.
+
+    Args:
+        gnps_network (string): path to tsv file generated from classical molecular.
+                               networking with study dataset(s) and reference dataset.
+        metadata (string): path to sample metadata (comma- or tab-separated) file.
+                           Must contain a column with mzXML file names that match those used in the molecular networking job.
+        filename_col (string): name of column header containing file names.
+        sample_types (string): one of 'simple', 'complex', or 'None'.
+                               Simple foods are single ingredients while complex foods contain multiple ingredients.
+                               'None' will return both simple and complex foods.
+        all_groups (list): can contain 'G1', 'G2' to denote study spectrum files.
+        some_groups (list): can contain 'G3', 'G4' to denote reference spectrum files.
+        level (integer): indicates the level of the food ontology to use.
+                         One of 1, 2, 3, 4, 5, or 6.
+        save_name (string): path/file name to save food counts table.
+    Return:
+        A data frame. The data frame is also saved as a csv file to the specified directory.
+    Examples:
+        get_dataset_food_counts(gnps_network = 'METABOLOMICS-SNETS-V2-07f85565-view_all_clusters_withID_beta-main.tsv',
+                                metadata = 'sample_metadata.csv',
+                                filename_col = 'filenames',
+                                sample_types = 'simple',
+                                all_groups = ['G1'],
+                                some_groups = ['G4'],
+                                level = 5,
+                                save_name = "food_counts_L5")
+    """
+    food_counts, filenames = [], []
+    gnps_network = pd.read_csv(gnps_network, sep='\t')
+    sample_types = get_sample_types(sample_types)
+    if metadata.endswith('.csv'): delim = ','
+    else: delim = '/t'
+    metadata = pd.read_csv(metadata, sep=delim)
+    metadata = metadata[metadata[filename_col].notnull()]
+    for filename in metadata[filename_col]:
+        file_food_counts = get_file_food_counts(gnps_network, sample_types, all_groups, some_groups, [filename], level)
+        if len(file_food_counts) > 0:
+            food_counts.append(file_food_counts)
+            filenames.append(filename)
+    food_counts = (pd.concat(food_counts, axis=1, sort=True).fillna(0).astype(int).T)
+    food_counts.index = pd.Index(filenames, name='filename')
+    food_counts.to_csv(save_name + '.csv')
```

Comment:
I'd avoid explicitly saving inside this function. The user can trivially do that themselves afterwards if they want to.